src/main.rs

```diff
@@ -512,3 +538,246 @@ fn setup() -> Opts {
+    /// * `pattern_from_file` - Boolean true = write 'regexp' to file (opts.file) false = write 'regexp' to String
+    /// * `output_path` - Either None or PathBuf to 'output.fq'
+    ///
+    /// # Examples
+    /// let seqs = vec![vec!["AAGTCTGAATCCATGGAAAGCTATTG", "GGGTCTGAATCCATGGAAAGCTATTG"], vec!["AAGTCTGAATCCATGGAAAGCTATTG", "GGGTCTGAATCCATGGAAAGCTATTG"]];
+    /// let pattern = Vec<String::from("AGTG")>;
+    /// let dir = = TempDir::new().unwrap();
+    /// let ex_opts = call_opts(dir, seqs, pattern, true)
+    /// ex_opts will be an instance of StructOpts where ex_opts.file is a PathBuf for a file with one line 'AGTG' and ex_opts.args is Vec<String> with two fastq paths
+    fn call_opts(
+        dir: &TempDir,
+        seqs: &Vec<Vec<&str>>,
+        regexp: &Vec<String>,
+        pattern_from_file: bool,
+        output: Option<PathBuf>,
+    ) -> Opts {
+        // write fastq
+        let fq_path = write_fastq(&dir, &seqs);
+
+        // define pattern as from file or from string (opts.file or opts.regexp)
+        if pattern_from_file {
+            // Write pattern.txt from provided patterns
+            let pattern_path: PathBuf = write_pattern(&dir, &regexp);
+            let regex: Vec<String> = vec![];
+            // Write instance of opts
+            let file_test_opts = write_opts(Some(pattern_path), &fq_path, &regex, output);
+            file_test_opts
+        } else {
+            // Call write_opts
+            let nofile_test_opts: Opts = write_opts(None, &fq_path, &regexp, output);
+            nofile_test_opts
+        }
+    }
+
+    /// Returns an instance of StructOpts within call_opts().
+    ///
+    /// # Arguments
+    ///
+    /// * `pattern_file_path` - Either None or PathBuf to 'pattern.txt' created in write_pattern()
+    /// * `fastq_file_path` - &Vec<String> created in write_fastq()
+    /// * `regexp_str` - Either an empty Vec<String> or Vec<String> where String = reg expressions
+    /// * `output_path` - Either None or PathBuf to 'output.fq'
+    ///
+    fn write_opts(
+        pattern_file_path: Option<PathBuf>,
+        fastq_file_path: &Vec<String>,
+        regex_str: &Vec<String>,
+        output_path: Option<PathBuf>,
+    ) -> Opts {
+        // Takes a PathBuf as Opts.file, &Vec<String> for Opts.args, and &Vec<String> for Opts.regexp
+        let return_opts = Opts {
+            // todo count must be false when in combination with an output path
+            threads: 4,
+            color: Color::Never,
+            count: false,
+            regexp: regex_str.to_vec(),
+            fixed_strings: false,
+            file: pattern_file_path,
+            invert_match: false,
+            decompress: false,
+            paired: false,
+            reverse_complement: false,
+            progress: true,
+            args: fastq_file_path.to_vec(),
+            output: output_path,
+        };
+        return_opts
+    }
+
```

Comment:
Add function to return sequences from fastq using seq_io::fastq
```suggestion
    /// Returns sequences from fastq
    /// 
    /// # Arguments
    /// * result_path" path to fastq of matches
    /// 
    /// # Example
    /// let dir: TempDir = TempDir::new().unwrap();
    /// let out_path = dir.path().join(String::from("output.fq"));
    /// // call fqgrep with opts.output = out_path
    /// let return_seq = slurp_output(out_path);
    fn slurp_output(result_path: PathBuf) -> Vec<String> {
        let mut return_seqs = vec![];
        let mut reader = fastq::Reader::from_path(result_path).unwrap();
        while let Some(record) = reader.next() {
            let record = record.expect("Error reading record");
            // convert bytes to str
            let seq = std::str::from_utf8(record.seq()).unwrap();
            return_seqs.push(seq.to_owned());
        }
        return_seqs
    }

```